modules/ascat/main.nf

```diff
@@ -0,0 +1,155 @@
+
+    input:
+    tuple val(meta), path(normal_bam), path(normal_bai), path(tumor_bam), path(tumor_bai)
+    path(allele_files)
+    path(loci_files)
+
+    output:
+    tuple val(meta), path("*png"),               emit: png
+    tuple val(meta), path("*cnvs.txt"),          emit: cnvs
+    tuple val(meta), path("*purityploidy.txt"),  emit: purityploidy
+    tuple val(meta), path("*segments.txt"),      emit: segments
+    path "versions.yml",                         emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args           = task.ext.args        ?: ''
+    def prefix         = task.ext.prefix      ?: "${meta.id}"
+    def gender         = args.gender          ?  "$args.gender" :        "NULL"
+    def genomeVersion  = args.genomeVersion   ?  "$args.genomeVersion" : "hg19"
+    def purity         = args.purity          ?  "$args.purity" :        "NULL"
+    def ploidy         = args.ploidy          ?  "$args.ploidy" :        "NULL"
+    def gc_files       = args.gc_files        ?  "$args.gc_files" :      "NULL"
+
+    def minCounts_arg                    = args.minCounts                     ?  ",minCounts = $args.minCounts" : ""
+    def chrom_names_arg                  = args.chrom_names                   ?  ",chrom_names = $args.chrom_names" : ""
+    def min_base_qual_arg                = args.min_base_qual                 ?  ",min_base_qual = $args.min_base_qual" : ""
+    def min_map_qual_arg                 = args.min_map_qual                  ?  ",min_map_qual = $args.min_map_qual" : ""
+    def ref_fasta_arg                    = args.ref_fasta                     ?  ",ref.fasta = '$args.ref_fasta'" : ""
+    def skip_allele_counting_tumour_arg  = args.skip_allele_counting_tumour   ?  ",skip_allele_counting_tumour = $args.skip_allele_counting_tumour" : ""
+    def skip_allele_counting_normal_arg  = args.skip_allele_counting_normal   ?  ",skip_allele_counting_normal = $args.skip_allele_counting_normal" : ""
+
+
+
+    """
+    #!/usr/bin/env Rscript
+    library(RColorBrewer)
+    library(ASCAT)
+    options(bitmapType='cairo')
+
+
+    #prepare from BAM files
+    ascat.prepareHTS(
+        tumourseqfile = "$tumor_bam",
+        normalseqfile = "$normal_bam",
+        tumourname = "Tumour",
+        normalname = "Normal",
+        allelecounter_exe = "alleleCounter",
+        alleles.prefix = "$allele_files/G1000_alleles_${genomeVersion}_chr",
+        loci.prefix = "$loci_files/G1000_loci_${genomeVersion}_chr",
+        gender = "$gender",
+        genomeVersion = "$genomeVersion",
+        nthreads = $task.cpus
+        $minCounts_arg
+        $chrom_names_arg
+        $min_base_qual_arg
+        $min_map_qual_arg
+        $ref_fasta_arg
+        $skip_allele_counting_tumour_arg
+        $skip_allele_counting_normal_arg
+    )
+
+
+    #Load the data
+    ascat.bc = ascat.loadData(
+        Tumor_LogR_file = "Tumour_tumourLogR.txt",
```

Comment:
It's running like it is in[ current Sarek-code](https://github.com/nf-core/sarek/blob/7ccfb36509d8380946c048065129b29d69c8443b/bin/run_ascat.r#L47). There it also changes name from 'normal' to 'germline'. Is that what you meant?

Reply:
I guess the naming is just not super obvious to me, so tumor_normal would imply to me it is somatic calling (see Germline_LogR_file) and tumor_tumor I guess would still have a single tumor, sample right? Ideally we could pass the sample IDs but I don't know this can easily be made generic.